rustls/tests/unbuffered.rs

```diff
@@ -870,6 +922,39 @@ fn write_traffic<T: SideData, R, F: FnMut(WriteTraffic<T>) -> R>(
     }
 }
 
+fn read_traffic<T: SideData, R, F: FnMut(ReadTraffic<T>) -> R>(
+    status: UnbufferedStatus<'_, '_, T>,
+    mut f: F,
+) -> (R, usize) {
+    let UnbufferedStatus { discard, state } = status;
+    let state = state.unwrap();
+    if let ConnectionState::ReadTraffic(state) = state {
+        (f(state), discard)
+    } else {
+        panic!("unexpected client state {state:?} (wanted ReadTraffic)");
+    }
+}
+
+fn peer_closed<T: SideData>(status: UnbufferedStatus<'_, '_, T>) -> usize {
+    let UnbufferedStatus { discard, state } = status;
+    let state = state.unwrap();
+    if let ConnectionState::PeerClosed = state {
+        discard
+    } else {
+        panic!("unexpected client state {state:?} (wanted PeerClosed)");
+    }
+}
+
+fn closed<T: SideData>(status: UnbufferedStatus<'_, '_, T>) -> usize {
```

Comment:
it seemed like `closed`, `peer_closed` and `read_traffic` could be made into one helper, but in practice because `ConnectionState` doesn't impl `PartialEq` the contortions I had to make to get it working didn't seem like a net improvement.